### PR TITLE
Make the errors of Sylo groups reusable

### PR DESCRIPTION
**Pull request by aliXsed**
I needed to reuse some errors of this module for the Sylo migration functions too. So I made them reusable as the first step.

### CODE REVIEW

**Review by jordy25519 — APPROVED** [review]
lgtm 👍

File: crml/sylo/src/groups/mod.rs [hunk 1]
```diff
@@ -99,6 +99,33 @@ where
...
 }
 
+decl_error! {
+	pub enum Error for Module<T: Trait> {
+		/// Group already exists
+		GroupAlreadyExists,
+		/// Already a member of the group
+		MemberAlreadyExists,
+		/// Invite already exists
+		InviteAlreadyExists,
```
Comment (jordy25519) [inline crml/sylo/src/groups/mod.rs]: I'd remove `Already` from these errors

File: crml/sylo/src/groups/mod.rs [hunk 2]
```diff
@@ -99,6 +99,33 @@ where
...
+		/// Can not invite more than maximum amount
+		MaxInvitesReached,
+		/// Can not store more than maximum number of members
+		MaxNumberOfMembersReached,
```
Comment (jordy25519) [inline crml/sylo/src/groups/mod.rs]: `MaxMembersReached`

**Review by jordy25519 — COMMENTED** [review]
Some suggestions to make the error name simpler

**Review by cowboy-bebug — APPROVED** [review]
Thanks for adding the errors! LGTM

File: crml/sylo/src/groups/mod.rs (hunk 1 above)
Comment (aliXsed) [reply crml/sylo/src/groups/mod.rs]: Done

File: crml/sylo/src/groups/mod.rs (hunk 2 above)
Comment (aliXsed) [reply crml/sylo/src/groups/mod.rs]: Done

**Pull request merged**